tests/cupy_tests/linalg_tests/test_product.py

```diff
@@ -71,6 +71,30 @@ def test_dot_with_out(self, xp, dtype_a, dtype_b, dtype_c):
         return c
 
 
+@testing.parameterize(*testing.product({
+    'shape': [
+        ((1, 2), (1, 2)),
+        ((1, 3), (1, 3)),
+        ((2, 2), (2, 2)),
+        ((2, 3), (2, 3)),
+        ((1, 2), (2, 2)),
+        ((1, 3), (2, 3)),
+        ((1, 2), (3, 2)),
+        ((1, 3), (3, 3)),
```

Comment:
Please organize the test patterns and put a brief comment on each of them (or on each group of them).
It looks some patterns are duplicated (for example, there's no point of testing both `((1, 3), (2, 3))` and `((1, 3), (3, 3))`).

Also there seem some test patterns that are missing, for example:
* The dimensions of the vectors are different (e.g. `((2, 2), (2, 3))` and vice versa)
* More than two batch dimensions (e.g. `((2, 4, 5, 3), (2, 4, 5, 3)`)
* No batch dimension (e.g. `((3,), (3,))`)
* 0 in the batch dimensions (e.g. `((0, 3), (0, 3)` and `(2, 0, 3) and (2, 0, 3)`)

Reply:
Okay alright I'll remove redundant ones and add additional test cases